### PR TITLE
Add the CMS Control interface schema.

### PR DESCRIPTION
**Pull request by Ciuffi**
I want to be able to turn the CMS off through firebase. This adds that functionality to the Database.

Here is the PR for the CMS changes: https://github.com/nwplus/CMS-V2/pull/42

### CODE REVIEW

File: nw-firestore-schema.ts [hunk 1]
```diff
@@ -101,7 +101,17 @@ interface FAQ {
   lastModifiedBy: string;
 }
 
+interface CMS {
+  CMSOff: boolean;
+  OffUntilDate: timestamp;
```
Comment (ianmah) [inline nw-firestore-schema.ts]: nit: camel case?

File: nw-firestore-schema.ts [hunk 2]
```diff
@@ -101,7 +101,17 @@ interface FAQ {
   lastModifiedBy: string;
 }
 
+interface CMS {
+  CMSOff: boolean;
```
Comment (ianmah) [inline nw-firestore-schema.ts]: nit: `status`, `cmsStatus`, `maintenanceMode`  ? sorry im being rly picky

File: nw-firestore-schema.ts [hunk 3]
```diff
@@ -101,7 +101,17 @@ interface FAQ {
...
+  OffUntilDate: timestamp;
+}
+
+interface InternalWebsites {
```
Comment (ianmah) [inline nw-firestore-schema.ts:108]: is this/should this be a collection?

**Comment by ianmah** [comment]
Not sure if this is better or worse:
`offUntilDate`: `false` => CMS is `On`
`offUntilDate`: `[date passed]` => CMS is `On`
`offUntilDate`: `[date upcoming]` => CMS is `Off`

**Comment by Ciuffi** [comment]
Its not a bad idea but i'm not sure if we can set it to false? Well I guess we could but its a bit strange to switch the type in the db. We could just set it to an old date for now?

**Comment by ianmah** [comment]
is `null` valid?

**Comment by Ciuffi** [comment]
I can't set it as null in the firebase interface, and it definitely complains when sending a null value

**Comment by Ciuffi** [comment]
Although the firebase UI doesn't mind switching the type whenever

**Comment by Ciuffi** [comment]
I think false | timestamp is fine then

File: nw-firestore-schema.ts (hunk 3 above)
Comment (Ciuffi) [reply nw-firestore-schema.ts:108]: I think its fine for it to be a collection, we might want to add feature flags for our other internal projects too.

**Comment by Ciuffi** [comment]
@ianmah I've decided to go with this schema instead, since I don't think its a good idea to automatically turn the CMS back on.

offUntilDate === false: CMS is on
offUntilDate === true: CMS is off with no date for when its back up
typeof offUntilDate === timestamp: CMS is off with the relaunch date present.

Thoughts?

**Comment by ianmah** [comment]
@Ciuffi are we talking about a firebase timestamp or a date string? Otherwise sounds good

**Comment by Ciuffi** [comment]
@ianmah using a firebase timestamp (mostly to simplify input)

**Comment by ianmah** [comment]
@Ciuffi I am apprehensive of using both  timestamps and timestamp strings in our code

**Comment by Ciuffi** [comment]
@ianmah  Are we using timestamp strings? So far I've only seen firebase timestamps in the CMS.

**Comment by ianmah** [comment]
According to schema:
https://github.com/nwplus/database-schemas/blob/9cf5c7365551c65feb5502b1f7e041fb4dda86aa/nw-firestore-schema.ts#L31
https://github.com/nwplus/database-schemas/blob/9cf5c7365551c65feb5502b1f7e041fb4dda86aa/nw-firestore-schema.ts#L54-L55
https://github.com/nwplus/database-schemas/blob/9cf5c7365551c65feb5502b1f7e041fb4dda86aa/nw-firestore-schema.ts#L101

**Comment by Ciuffi** [comment]
I checked the database and it looks like every single section is using a different timestamping method lol.

Intro uses an ISO string.

Sponsor uses a human readable string.

FAQ uses a firebase timestamp.

I think we should probably default to firebase timestamp. Originally I didn't want to use it because I thought the conversation would be a bit weird, but it seems to work fine.

I'll bring it up at the next meeting so we can make the change to firebase timestamp.

**Comment by ianmah** [comment]
Agreed, I am not particular to which we use, would just prefer all use the same one

**Comment by Ciuffi** [comment]
@ianmah I agree. Since we've decided to use firebase timestamp, could you approve this?

File: nw-firestore-schema.ts (hunk 3 above)
Comment (ianmah) [reply nw-firestore-schema.ts:108]: Yeah I wanted it to be a collection

Comment (Ciuffi) [reply nw-firestore-schema.ts:108]: it is a collection

Comment (ianmah) [reply nw-firestore-schema.ts:108]: oh, I thought it would have a structure more similar to `Hackathons` which is defined as ` Hackathon[]`

Comment (Ciuffi) [reply nw-firestore-schema.ts:108]: I'm not really sure what that structure will look like yet so right now I just chose to make one for the CMS

File: nw-firestore-schema.ts (hunk 1 above)
Comment (Ciuffi) [reply nw-firestore-schema.ts]: fixed

File: nw-firestore-schema.ts (hunk 2 above)
Comment (Ciuffi) [reply nw-firestore-schema.ts]: just using the date now

**Review by ianmah — APPROVED** [review]
Esketit

**Pull request merged**